amcat4apiclient/amcat4apiclient.py

```diff
@@ -17,17 +18,13 @@ def serialize(obj):
 
 
 class AmcatClient:
-    def __init__(self, host, username, password, ignore_tz=True):
+    def __init__(self, host, username=None, password=None, ignore_tz=True):
         self.host = host
         self.username = username
-        self.token = self.get_token(password)
+        self.password = password
         self.ignore_tz = ignore_tz
-
-    def get_token(self, password) -> str:
-        r = requests.post(self.url("auth/token"),
-                          data=dict(username=self.username, password=password))
-        r.raise_for_status()
-        return r.json()["access_token"]
+        # run at init to cache token
+        _get_token(self.host, self.username, self.password)
```

Comment:
It feels like we're doubting somehow between doing this object-oriented (in which case storing token, middlecat url etc on the object would make sense) or functional?

Reply:
There are two reasons why I thought this would be better left to functions instead of storing it in the object: 1. I wanted to keep the auth code separate since it is rather complex, imports a lot of stuff and would clutter the main amcat4apiclient.py; 2. we talked so much about the token being potentially very valuable that it made sense to me to hide it a little from the user (if they are savvy enough they can import `secret_read` and decode the token if they really want).

I'm not totally clear if it would make sense to have at least the functions that require interactivity be object-oriented. Only `get_middlecat_token`  does that but in that case it would probably make sense to also move `_get_token`, (`get_password_token`) and  `check_token` over. 

What do you think?